### PR TITLE
Logging in expert choice

### PR DESCRIPTION
**Pull request by ludziej**
# Description
- Fixed heavy logging in expert choice
- More timing in expert choice
- Reduced one permute
- Fixed configs

[Neptune](https://app.neptune.ai/o/pmtest/org/llm-random/runs/details?viewId=9a15aa36-1063-49c4-807d-53393abb118a&detailsTab=metadata&shortId=LLMRANDOM-4149&type=run&compare=CwBhCYBoEZgdiA&path=block_3%2Ffeedforward%2Ftime%2F&attribute=plot_0)

# Checklist
   - [x] Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - [x] Are all variables set with logical defaults that are backward compatible?
   - [x] I attached link to neptune if it was necessary
   - [x] Have you successfully executed the linter and tests?
   - [x] Are all functions concise and don't require splitting?
   - [x] Is there documentation for arguments and complex logic?
   - [x] Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - [x] Are all functions placed in appropriate files?
   - [x] Is the PR name meaningful and descriptive?
   - [x] Is PR description provided, or unnecessary?
  
Other questions:
   - [x] If I made changes to the requirements or anything related to the Singularity image I did generate new image and upload a new image to the clusters
   - [x] If this change is significant enough I notified all individuals working with this repository (@channel in slack channel)
   - [x] If there is a need for a second reviewer I requested additional review 

# Reviewer's checklist:
   - [ ] Does the Neptune link work and does it actually compare the situation before and after the PR?
   - [ ] Do I understand the code?
   - [ ] Do I think another reviewer is not needed or I commented that this PR need another reviewer?

### CODE REVIEW

File: research/conditional/train/configs/baselines/gpt/dense/large.yaml
Comment (Simontwice) [inline research/conditional/train/configs/baselines/gpt/dense/large.yaml:1]: this calls for a separate PR

**Review by Simontwice — CHANGES_REQUESTED** [review]
Please review the comments

File: research/conditional/utils/layer_manager.py [hunk 1]
```diff
@@ -40,7 +40,12 @@ def _register_layers(self, model):
             registered_name = None
             suffix = name.split(".")[-1]
 
-            if suffix in ["residual_feedforward", "residual_attention", "feedforward"]:
+            if suffix in [
+                "residual_feedforward",
+                "residual_attention",
+                "feedforward",
+                "expert_gating",
```
[the remark this answers is not in the document]
Comment (ludziej) [reply research/conditional/utils/layer_manager.py:47]: the thing is, here we need our another class "expert gating" not "expert choice" which is a submodule to be indexed by layer manager. This is primarly why heavy logging and time measuring did not work previously in Granular MoE - if class is not managed by layer manager, you cannot do any of these.

[the remark this answers is not in the document]
Comment (ludziej) [reply research/conditional/utils/layer_manager.py:47]: I think moving the list of "allowed" names to init should be a better solution.

[the remark this answers is not in the document]
Comment (Simontwice) [reply research/conditional/utils/layer_manager.py:47]: I understand now, thanks!

[the remark this answers is not in the document]
Comment (Simontwice) [reply research/conditional/utils/layer_manager.py:47]: I'm not sure whether we need to worry about such a refactor right now.

**Review by Simontwice — APPROVED** [review]
LGTM, but I'm no expert on expert choice

**Pull request merged**